runtime/runtime/src/lib.rs

```diff
@@ -1098,6 +1100,23 @@ impl Runtime {
                 &mut stats,
             )?;
         }
+        if !apply_state.is_new_chunk
+            && apply_state.current_protocol_version
+                >= ProtocolFeature::FixApplyChunks.protocol_version()
```

Comment:
This is unrelated to the PR, but we could (and maybe should) fix the API discrepancy between stabe & unstable features like this:

```rust
diff --git a/core/primitives/Cargo.toml b/core/primitives/Cargo.toml
index 92d31a613..438301be6 100644
--- a/core/primitives/Cargo.toml
+++ b/core/primitives/Cargo.toml
@@ -39,10 +39,8 @@ near-vm-errors = { path = "../../runtime/near-vm-errors" }
 near-rpc-error-macro = { path = "../../tools/rpctypegen/macro" }
 
 [features]
-default = ["jemallocator", "stable_protocol_features"]
+default = ["jemallocator"]
 dump_errors_schema = ["near-rpc-error-macro/dump_errors_schema"]
-stable_protocol_features = ["protocol_feature_lower_storage_cost"]
-protocol_feature_lower_storage_cost = []
 protocol_feature_add_account_versions = ["near-primitives-core/protocol_feature_add_account_versions"]
 protocol_feature_forward_chunk_parts = []
 protocol_feature_rectify_inflation = []
diff --git a/core/primitives/src/runtime/config.rs b/core/primitives/src/runtime/config.rs
index c3494725a..be67b0b32 100644
--- a/core/primitives/src/runtime/config.rs
+++ b/core/primitives/src/runtime/config.rs
@@ -60,7 +60,7 @@ impl RuntimeConfig {
         protocol_version: ProtocolVersion,
     ) -> Arc<Self> {
         if checked_feature!(
-            "protocol_feature_lower_storage_cost",
+            "stable",
             LowerStorageCost,
             protocol_version
         ) {
diff --git a/core/primitives/src/version.rs b/core/primitives/src/version.rs
index 8b2282e3c..c28dec1db 100644
--- a/core/primitives/src/version.rs
+++ b/core/primitives/src/version.rs
@@ -85,7 +85,6 @@ pub enum ProtocolFeature {
     #[cfg(feature = "protocol_feature_block_header_v3")]
     BlockHeaderV3,
     /// Decreases the storage cost of 1 byte by 10X.
-    #[cfg(feature = "protocol_feature_lower_storage_cost")]
     LowerStorageCost,
     #[cfg(feature = "protocol_feature_alt_bn128")]
     AltBn128,
@@ -111,7 +110,6 @@ impl ProtocolFeature {
     pub const fn protocol_version(self) -> ProtocolVersion {
         match self {
             // Stable features
-            #[cfg(feature = "protocol_feature_lower_storage_cost")]
             ProtocolFeature::LowerStorageCost => 42,
             ProtocolFeature::DeleteActionRestriction => 43,
             ProtocolFeature::FixApplyChunks => 44,
@@ -139,6 +137,10 @@ impl ProtocolFeature {
 
 #[macro_export]
 macro_rules! checked_feature {
+    ("stable", $feature:ident, $current_protocol_version:expr) => {{
+        $crate::version::ProtocolFeature::$feature.protocol_version() <= $current_protocol_version
+    }};
+
     ($feature_name:tt, $feature:ident, $current_protocol_version:expr) => {{
         #[cfg(feature = $feature_name)]
         let is_feature_enabled = $crate::version::ProtocolFeature::$feature.protocol_version()
diff --git a/runtime/runtime/src/lib.rs b/runtime/runtime/src/lib.rs
index 092dad9b2..68c5206d1 100644
--- a/runtime/runtime/src/lib.rs
+++ b/runtime/runtime/src/lib.rs
@@ -7,6 +7,7 @@ use log::debug;
 pub use near_crypto;
 use near_crypto::PublicKey;
 pub use near_primitives;
+use near_primitives::checked_feature;
 use near_primitives::runtime::get_insufficient_storage_stake;
 use near_primitives::{
     account::{AccessKey, Account},
@@ -1101,8 +1102,7 @@ impl Runtime {
             )?;
         }
         if !apply_state.is_new_chunk
-            && apply_state.current_protocol_version
-                >= ProtocolFeature::FixApplyChunks.protocol_version()
+            && checked_feature!("stable", FixApplyChunks, apply_state.current_protocol_version)
         {
             let (trie_changes, state_changes) = state_update.finalize()?;
             let proof = trie.recorded_storage();
```

I'll look into that after  this PR is merged

Reply:
@matklad Thanks!